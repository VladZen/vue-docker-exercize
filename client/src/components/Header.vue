<template>
  <div class="layout__header">
    <div v-if="$slots.buttons" class="header__btns">
      <slot name="buttons"></slot>
    </div>

    <h1>{{ merchantName }}<br />Inventory</h1>
    <slot></slot>
  </div>
</template>

<script>
import { mapState, mapActions } from "vuex";

export default {
  created() {
    this.fetchMerchantName();
  },
  computed: {
    ...mapState(["merchantName"]),
  },
  methods: {
    ...mapActions({
      fetchMerchantName: "GET_MERCHANT",
    }),
  },
};
</script>

<style lang="scss" scoped></style>
