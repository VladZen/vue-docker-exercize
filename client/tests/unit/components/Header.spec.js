import { mount, createLocalVue } from "@vue/test-utils";
import Header from "@/components/Header";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

const MERCHANT_NAME = "Some Merchant Name";

const store = new Vuex.Store({
  state: () => ({ merchantName: "" }),
  actions: {
    GET_MERCHANT({ state }) {
      state.merchantName = MERCHANT_NAME;
    },
  },
});

const mountHeader = (buttons = "") => {
  return mount(Header, {
    store,
    localVue,
    slots: {
      buttons,
    },
  });
};

describe("Header", () => {
  it("should call fetchMerchantName on initialize", () => {
    const wrapper = mountHeader();
    expect(wrapper.vm.merchantName).toBe(MERCHANT_NAME);
  });

  it("checks rendering without buttons", () => {
    const wrapper = mountHeader();
    expect(wrapper).toMatchSnapshot();
  });

  it("checks rendering with buttons", () => {
    const wrapper = mountHeader(`<button type="button">Some Button</button>`);
    expect(wrapper).toMatchSnapshot();
  });
});
