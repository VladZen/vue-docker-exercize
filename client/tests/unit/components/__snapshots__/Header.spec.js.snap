// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header checks rendering with buttons 1`] = `
<div class="layout__header">
  <div class="header__btns"><button type="button">Some Button</button></div>
  <h1>Some Merchant Name<br>Inventory</h1>
</div>
`;

exports[`Header checks rendering without buttons 1`] = `
<div class="layout__header">
  <!---->
  <h1>Some Merchant Name<br>Inventory</h1>
</div>
`;
